streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -83,9 +83,26 @@ public void register(final StateRestorer restorer) {
 
         final Set<TopicPartition> restoringPartitions = new HashSet<>(needsRestoring.keySet());
         try {
-            final ConsumerRecords<byte[], byte[]> allRecords = poll(restoreConsumer, 10);
-            for (final TopicPartition partition : restoringPartitions) {
-                restorePartition(allRecords, partition, active.restoringTaskFor(partition));
+            final Map<TopicPartition, Long> endOffsets = restoreConsumer.endOffsets(restoringPartitions);
```

Comment:
I am wondering, if this should be a class member that is update once at "restore begin" but not each time we check for the next "batch of records" ?